perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/pwdmgr/MuadmPasswordManagerModule.java

```diff
@@ -39,14 +46,29 @@ public void checkPasswordStrength(PerunSession sess, String login, String passwo
 			throw new PasswordStrengthException("Password for " + actualLoginNamespace + ":" + login + " cannot be empty.");
 		}
 
-		if (password.length() < 10) {
-			log.warn("Password for {}:{} is too short. At least 10 characters are required.", actualLoginNamespace, login);
-			throw new PasswordStrengthException("Password for " + actualLoginNamespace + ":" + login + " is too short. At least 10 characters is required.");
+		if (password.length() < 14) {
+			log.warn("Password for {}:{} is too short. At least 14 characters are required.", actualLoginNamespace, login);
+			throw new PasswordStrengthException("Password for " + actualLoginNamespace + ":" + login + " is too short. At least 14 characters is required.");
 		}
 
-		if (!StringUtils.isAsciiPrintable(password)) {
-			log.warn("Password for {}:{} must contain only printable characters.", actualLoginNamespace, login);
-			throw new PasswordStrengthException("Password for " + actualLoginNamespace + ":" + login + " must contain only printable characters.");
+		if (login != null) {
```

Comment:
Just to be sure, this check should be performed only if the longer is 3 or more chars longer (this is taken from the specifiaction in AD).
```suggestion
		if (login != null && login.length() > 2) {
```